### PR TITLE
Added gold and melas codes and modified alternant codes

### PR DESCRIPTION
**Pull request by khalilguy**

### CODE REVIEW

File: codes/classical/bits/gold.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+# Use UTF-8 unicode encoding
```
Comment (valbert4) [inline codes/classical/bits/gold.yml:6]: Please remove all comments that contain directions.

File: codes/classical/bits/gold.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+code_id: gold
+physical: bits
+logical: bits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title (SINGULAR) + first reference(s) (optional).
+name: 'Gold'
+introduced: '\cite{doi:10.1109/TIT.1967.1054048}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
```
Comment (valbert4) [inline codes/classical/bits/gold.yml:29]: Throughout the entry: symbols r, m, x, y etc should be in LaTeX.

File: codes/classical/bits/gold.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# that entry instead of here.
+description: 'A \([2^r -1, 2r ]\) is a cyclic linear code characterized by the generator polynomial of degree r of a preferred-pair of m-sequences.'
+
+features:
+  
+  encoders:
+    - 'Codewords of Gold codes are generated using m-sequences x and y, which are codewords of simplex codes with check polynomials of degree r. The information bits are initialized in the shift registers of the the two m-sequences x and y.'
```
Comment (valbert4) [inline codes/classical/bits/gold.yml:36]: This should be moved to description. "Encoders" is more for efficient algorithms to encode for particular codes, mostly quantum.

File: codes/classical/bits/gold.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  decoders:
+    - 'General decoding is done by building a sparse parity check matrix, followed by applying an iterative message passing alogirithm. \cite{doi:10.1007/s11265-006-7278-y}, \cite{doi:10.1109/18.910572}.'
+
+  # Propose a field that you feel is applicable to a large and interesting class of codes
+  #some_other_field:
+  # - 'detail'
+
+realizations:
+  - 'Used in for synchronization purposes in telecommunication and satelite navigation.'
```
Comment (valbert4) [inline codes/classical/bits/gold.yml:46]: Would be great to have a citation here, but I understand finding these can be hard.

File: codes/classical/properties/alternant.yml [hunk 5]
```diff
@@ -10,8 +10,19 @@ logical: q-ary_digits
 name: 'Alternant code'
 introduced: '\cite{doi:10.1016/S0019-9958(74)80005-7}'
 
-description: 'Stub.'
+description: 'A code with parity check matrix with n columns and t roww of the form
+\( H = \begin{bmatix}
```
Comment (valbert4) [inline codes/classical/properties/alternant.yml:14]: Can be made into displayed eqn using   
  \begin{align}
  \begin{split}
  eqn\\
  eqn.
  \end{split}
  \end{align}
See shor_nine.yml for details.

File: codes/classical/properties/alternant.yml [hunk 6]
```diff
@@ -10,8 +10,19 @@ logical: q-ary_digits
...
+y_1g_2(x_1) & y_2g_2(x_2) & \hdots & y_ng_2(x_n)\\
+vdots & \vdots & & \vdots\\
+y_1g_t(x_1) & y_2g_t(x_2) \hdots y_ng_t(x_n)\\
+\end{bmatix} \) where each \(y_i \in F_{q^m}\), g is a polynomial with coefficents in \(F_{q^m}\)
+with degree \(\leq t-1\), and each \(x_i\) is a distinct element in \(F_{q^m}\).'
 
+#features:
+
+realizations:
+  - 'Used in compact disks, DVDs, two-dimensional bar codes, and satelitte communication'
```
Comment (valbert4) [inline codes/classical/properties/alternant.yml:25]: Citations would be great, esp in DOI form.

File: codes/classical/properties/alternant.yml [hunk 7]
```diff
@@ -10,8 +10,19 @@ logical: q-ary_digits
 name: 'Alternant code'
 introduced: '\cite{doi:10.1016/S0019-9958(74)80005-7}'
 
-description: 'Stub.'
+description: 'A code with parity check matrix with n columns and t roww of the form
```
Comment (valbert4) [inline codes/classical/properties/alternant.yml:13]: This is a large code class AFAIK; maybe explain why it's useful to consider this as a class. What are the properties, besides the defn, that unite all the codes in this class? Is there an efficient decoder?

File: codes/classical/bits/gold.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+logical: bits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title (SINGULAR) + first reference(s) (optional).
+name: 'Gold'
+introduced: '\cite{doi:10.1109/TIT.1967.1054048}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'A \([2^r -1, 2r ]\) is a cyclic linear code characterized by the generator polynomial of degree r of a preferred-pair of m-sequences.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/gold.yml:31]: Replace "preferred-pair of m-sequences." with "two maximum-period sequences of period 2^r-1 with absolute cross-correlation ≤ 2^((r+2)/2)."

File: codes/classical/bits/gold.yml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+features:
+  
+  encoders:
+    - 'Codewords of Gold codes are generated using m-sequences x and y, which are codewords of simplex codes with check polynomials of degree r. The information bits are initialized in the shift registers of the the two m-sequences x and y.'
+
+  decoders:
+    - 'General decoding is done by building a sparse parity check matrix, followed by applying an iterative message passing alogirithm. \cite{doi:10.1007/s11265-006-7278-y}, \cite{doi:10.1109/18.910572}.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/gold.yml:39]: The second of these references doi:10.1109/18.910572 (Factor graphs and the sum product algorithm) does not quite apply. 
Add reference to R. Gold's original paper https://doi.org/10.1109/TIT.1967.1054048

File: codes/classical/bits/gold.yml (hunk 8 above)
Comment (AlexanderBarg) [inline codes/classical/bits/gold.yml:31]: Add reference to R. Gold's original paper https://doi.org/10.1109/TIT.1967.1054048

File: codes/classical/bits/melas.yml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+# Code title (SINGULAR) + first reference(s) (optional).
+name: 'Melas'
+introduced: '\cite{doi:10.1070/RM1997v052n06ABEH002155}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'It is a cyclic linear code with generator polynomial is \(g(x) = p(x)p(x)*\), where \(p(x)\) is a primitive polynomial of degree m, being the minimal polynomial over \(F_2\) of an element \(\alpha\) of order \(2^m -1\) in \(F_{2^m}\), m is odd \(\geq 5 \) and * denotes reciprocation \cite{doi:10.1007/s12095-015-0135-8}. The parameters of this code are \([2^m -1, 2^m - 1 - 2m, 5]\).'
+
+#features:
+relations:
```
Comment (AlexanderBarg) [inline codes/classical/bits/melas.yml:34]: Kasami codes, cyclic codes, quaternary cyclic codes

**Pull request closed without merging**